content/src/instrument.ts

```diff
@@ -0,0 +1,24 @@
+import { AppComponents } from './types'
+
+export function instrumentWithQueryDurationMetric<F extends (...args: any[]) => any>(
+  components: Pick<AppComponents, 'metrics'>,
+  queryNameLabel: string,
+  functionToInstrument: F
+): F {
+  return <F>async function (...args: any[]) {
+    const { end: endTimer } = components.metrics.startTimer('dcl_db_query_duration_seconds', {
+      query: queryNameLabel
+    })
+    let queryStatus = 'success'
+    return new Promise<any>((res) => {
+      res(functionToInstrument(...args))
+    })
+      .catch((err) => {
+        queryStatus = 'error'
+        throw new Error(err)
+      })
+      .finally(() => {
+        endTimer({ status: queryStatus })
+      })
```

Comment:
May I suggest a more readable version?

```suggestion
    try {
       const res = await functionToInstrument(...args)
       endTimer({ status: 'success' })
       return res
    } catch (err) {
      throw err
    }
    return new Promise<any>((res) => {
      res(functionToInstrument(...args))
    })
      .catch((err) => {
        queryStatus = 'error'
        throw new Error(err)
      })
      .finally(() => {
        endTimer({ status: queryStatus })
      })
```

Reply:
appreciated